lib/sqlsync-solid-js/src/context.tsx

```diff
@@ -0,0 +1,34 @@
+import { SQLSync } from "@orbitinghail/sqlsync-worker";
+import { ParentComponent, createContext, createSignal, onCleanup } from "solid-js";
+
+export const SQLSyncContext = createContext<[() => SQLSync | null, (sqlSync: SQLSync) => void]>([
+  () => null,
+  () => {},
+]);
+
+interface Props {
+  workerUrl: string | URL;
+  wasmUrl: string | URL;
+  coordinatorUrl?: string | URL;
+}
+
+export const createSqlSync = (props: Props): SQLSync => {
+  return new SQLSync(props.workerUrl, props.wasmUrl, props.coordinatorUrl);
+};
+
+export const SQLSyncProvider: ParentComponent<Props> = (props) => {
+  const [sqlSync, setSQLSync] = createSignal<SQLSync>(createSqlSync(props));
```

Comment:
I'm not super familiar with how SolidJS works, but it appears that every time this component (SQLSyncProvider) renders it will re-create the SQLSync object. This is undesirable as each instance of the SQLSync object will initialize a connection to the shared worker at construction time. Is it possible to only call createSqlSync once if the signal has not yet been set?

Reply:
Solid JS calls component functions once, so this should correctly create the SQLSync object once. 

See here for more info: https://www.solidjs.com/guides/getting-started#2-vanishing-components